Tensile/SolutionStructs.py

```diff
@@ -3458,6 +3459,145 @@ def assignDerivedParameters(state):
+        else:
+          # for 1LDSB, we have to issue localwrites after localreads
+          # we have enough vgprBuffer to schedule localRead in the front of loop
+          if numVgprBuffer == state["LoopIters"]:
+            # fp16 or bf16, we read 1 element to vgprBuffer the other element to tempVgpr.
+            # since each iteration shares same tempVgpr, only read-to-vgprBuffer can 
+            # be scheduled in the front of loop.
+            # localwrite have to start after last read-to-tempVgpr.
+            if numReadPerVectorA != 1 or numReadPerVectorB !=1:
+              numHalfReads = (numReadPerVectorA//2)*state["InnerUnroll"]*state["MIWaveTileA"] + (numReadPerVectorB//2)*state["InnerUnroll"]*state["MIWaveTileB"]
+              numMfmaForHalfRead = 1
+              latencyLeft = miLatencyLeft
+              for i in range(numHalfReads):
+                latencyLeft -= 2
+                if latencyLeft < 0:
+                  numMfmaForHalfRead += 1
+                  latencyLeft = max(miLatencyLeft - 2, 0)
+              lwStartMfmaIndex = numMfmaPerIter * (state["LoopIters"] - 1 - numItersPLR) + numMfmaForHalfRead
+            else:
+              numReads = (numReadsPerIterA+numReadsPerIterB) * (state["LoopIters"]//(state["LocalReadVectorWidth"]//state["MIInputPerThread"]) - numItersPLR)
+              numMfmaForCurrentLoopLR = 1
+              latencyLeft = miLatencyLeft
+              for i in range(numReads):
+                latencyLeft -= issueLatencyB*2
+                if latencyLeft < 0:
+                  numMfmaForCurrentLoopLR += 1
+                  latencyLeft = max(miLatencyLeft - issueLatencyB*2,0)
+              if state["MIWaveTileA"] * state["MIWaveTileB"] > 1:
+                numMfmaForCurrentLoopLR += 1
+              lwStartMfmaIndex = numMfmaForCurrentLoopLR
+          else:
+            lwStartMfmaIndex = numMfmaPerIter * (state["LoopIters"] - 1 - numItersPLR) + numMfmaForLR
+          # to calculate number of mfma we need to wait before data arrive from lds to vgpr.
+          # latency: 40 quad-cycle for 4 word, 20 quad-cycle for 2 word, 10 quad-cycle for 1 word / half word
+          latencyForLR = roundUp(blockWidthB)*10
+          latencyForLR -= max(latencyLeft,0) # remaining latency in mfma
+          while latencyForLR > 0:
+            latencyForLR -= miLatency
+            lwStartMfmaIndex += 1
+        #########
+        # Get localWritePerMfma
+        #########
+        temp1 = 0
+        temp2 = 100
+        writesToSched = (numLoadsA+numLoadsB-1)*100
+        loop = 0
+        if lwStartMfmaIndex > lwEndMfmaIndex:
+          lwStartMfmaIndex = lwEndMfmaIndex
+        numMfmaCanSched = lwEndMfmaIndex - lwStartMfmaIndex + 1
+        while temp1 != temp2 and loop < 10:
+          loop += 1
+          temp1 = temp2
+          temp2 = roundUp((writesToSched+1 + temp1 + temp1%100 - (writesToSched+1) % temp1)/numMfmaCanSched)
+        state["LocalWritePerMfma"] = temp2/100
```

Comment:
Duplicated

Reply:
duplicated part in SolutionStructs.py is used for developing. It shows LWPM in kernel name.